tools-cli/src/main/java/com/hedera/hashgraph/client/cli/options/CollateCommand.java

```diff
@@ -226,47 +221,69 @@ private List<List<String>> verifyTransactions() {
 			// ensuring the required signatures are present.
-			helper.collate(infoFiles);
+			try {
+				helper.collate(infoFiles);
+			} catch (HederaClientRuntimeException e) {
+				// If collating failed, add an item to the verification list
+				// Make sure this transaction isn't already in the map
+				// (Multi-node submission will have duplicate transactionId entries)
+				if (!verifyWithFiles.containsKey(transactionId)) {
+					var verificationItemList = new ArrayList<String>();
+					verificationItemList.add(fileName);
+					verificationItemList.add(transactionId);
+					verifyWithFiles.put(transactionId, verificationItemList);
+					logger.info("Collation and Verification of " + transactionId
+							+ " failed due to the following error: "
+							+ e.getMessage().replace("Hedera Client Runtime: ", ""));
+				}
+				iterator.remove();
+				continue;
+			}
 
 			// Get the accounts associated with the transaction. This would include
 			// the fee payer, and accounts to be updated, or accounts with balances changing
 			// due to transfer, etc.
-			var requiredIds = helper.getSigningAccounts().stream()
+			final var accounts = helper.getSigningAccounts();
+			final var requiredIdsInUse = accounts.stream()
+					.filter(account -> !java.util.Arrays.stream(infoFiles)
+							.filter(a -> CommonMethods.getInfoFiles(a, account).length > 0)
+							.collect(Collectors.toList())
+							.isEmpty())
 					.map(AccountId::toString)
 					.collect(Collectors.toList());
 
 			// Get the list of public key names used to sign the transaction (if the key is a required key)
 			var publicKeyNames = getPublicKeyNames(helper);
 			// Sort the list of ids
-			Collections.sort(requiredIds);
+			Collections.sort(requiredIdsInUse);
 			// Sort the list of public keys
 			Collections.sort(publicKeyNames);
 
 			// Now put everything into the list for the csv
 			// transactionFileName, transactionId, list of accounts (requiredIds), list of keys used (getPublicKeyNames)
-			var verificationItemList = new ArrayList<String>();
-			verificationItemList.add(fileName);
-			verificationItemList.add(transactionId);
-			verificationItemList.add("\"" + String.join(",", requiredIds) + "\"");
-			verificationItemList.add("\"" + String.join(",", publicKeyNames) + "\"");
-
-
-			// Put the list of strings into the map
-			verifyWithFiles.put(transactionId, verificationItemList);
+			// Make sure this transaction isn't already in the map
+			// (Multi-node submission will have duplicate transactionId entries)
+			if (!verifyWithFiles.containsKey(transactionId)) {
+				var verificationItemList = new ArrayList<String>();
+				verificationItemList.add(fileName);
+				verificationItemList.add(transactionId);
+				verificationItemList.add("\"" + String.join(",", requiredIdsInUse) + "\"");
+				verificationItemList.add("\"" + String.join(",", publicKeyNames) + "\"");
+
+				// Put the list of strings into the map
+				verifyWithFiles.put(transactionId, verificationItemList);
+			}
```

Comment:
Seems this logic can be in a extracted to a method ? Seems almost similar to logic on 239

Reply:
True, and updated.